tests/test_save_as_python.py

```diff
@@ -0,0 +1,64 @@
+import tempfile
+
+import numpy as np
+import pytest
+from addict import Dict
+
+from cadet import Cadet
+
+
+@pytest.fixture
+def temp_cadet_file():
+    """
+    Create a new Cadet object for use in tests.
+    """
+    model = Cadet()
+
+    with tempfile.NamedTemporaryFile() as temp:
+        model.filename = temp
+        yield model
+
+
+def test_save_as_python(temp_cadet_file):
+    """
+    Test that the Cadet class raises a KeyError exception when duplicate keys are set on it.
+    """
+    # initialize "sim" variable to be overwritten by the exec lines later
+    sim = Cadet()
+
+    # Populate temp_cadet_file with all tricky cases currently known
+    temp_cadet_file.root.input.foo = 1
+    temp_cadet_file.root.input.bar.baryon = np.arange(10)
+    temp_cadet_file.root.input.bar.barometer = np.linspace(0, 10, 9)
+    temp_cadet_file.root.input.bar.init_q = np.array([], dtype=np.float64)
+    temp_cadet_file.root.input["return"].split_foobar = 1
+
+    code_lines = temp_cadet_file.save_as_python_script(filename="temp.py", only_return_pythonic_representation=True)
+
+    # remove code lines that save the file
+    code_lines = code_lines[:-2]
+
+    # populate "sim" variable using the generated code lines
+    for line in code_lines:
+        exec(line)
```

Comment:
Tbh, I'm not entirely sure about exec either.

Reply:
Sorry for not being more constructive, it was quite a busy day. I can have another look tomorrow.